### PR TITLE
platform_version compatibility mode

### PR DESCRIPTION
**Pull request by sersut**
This PR is the chef-ingredient side of https://github.com/chef/mixlib-install/pull/107. 

/cc: @chef-cookbooks/engineering-services

### CODE REVIEW

**Comment by yzl** [comment]
👍 pending travis

File: .kitchen.yml [hunk 1]
```diff
@@ -36,7 +37,7 @@ default: &default
 suites:
   - name: default
```
Comment (schisamo) [inline .kitchen.yml]: This file is feeling over-complicated...do we need all this includes/excludes logic still?

File: libraries/chef_ingredient_resource.rb [hunk 2]
```diff
@@ -41,6 +41,10 @@ class ChefIngredient < Chef::Resource::LWRPBase
 
       # Attribute to accept the license when applicable
       attribute :accept_license, kind_of: [TrueClass, FalseClass], default: false
+
+      # Attribute to enable selecting packages built for earlier versions in
+      # platforms that are not yet officially added to Chef support matrix
+      attribute :platform_version_backward_compat_mode, kind_of: [TrueClass, FalseClass], default: false
```
Comment (schisamo) [inline libraries/chef_ingredient_resource.rb]: It sounds like the group is moving toward consensus on `platform_version_compatability_mode` as the option name over in mixlib-install: https://github.com/chef/mixlib-install/pull/107#issuecomment-214858738

If that ends up being the case we should use the same attribute name here.

File: .kitchen.yml (hunk 1 above)
Comment (sersut) [inline .kitchen.yml]: Yeah. Some suites need to run on certain platforms. 

Having all suites on all platforms is a no-go. But I would want to cut down the number of platforms we run each suite on. For e.g. run only one suite on ubuntu-1204 and other suite on ubuntu-1004. 

@patrick-wright thoughts on that?

File: libraries/chef_ingredient_resource.rb (hunk 2 above)
Comment (sersut) [inline libraries/chef_ingredient_resource.rb]: Yep will do.

File: .kitchen.yml (hunk 1 above)
Comment (sersut) [inline .kitchen.yml]: Since Patrick is out I will probably merge this as it is and discuss the matrix when he is back.

**Comment by sersut** [comment]
OK so this is ready to go. I have fixed the option name and tested this with the linked mixlib-install PR:

```
export MIXLIB_INSTALL_GIT_REF=f0e23dfe396432502bbff11579979d1902f57d17
```

@schisamo let's chat tomorrow to ensure delivery-cluster scenario is fixed and we can get this out.

**Pull request merged**